cli/commands/preview/lib/appdata.ts

```diff
@@ -0,0 +1,132 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { z } from 'zod';
+import { LoggerError } from 'cli/logger';
+
+const SnapshotSchema = z.object( {
+	url: z.string(),
+	atomicSiteId: z.number(),
+	localSiteId: z.string(),
+	date: z.number(),
+	name: z.string().optional(),
+	userId: z.number().optional(),
+} );
+
+const UserDataSchema = z.object( {
+	version: z.number().optional(),
+	sites: z.array( z.any() ).optional(),
+	snapshots: z.array( SnapshotSchema ).optional(),
+	authToken: z.any().optional(),
+	locale: z.string().optional(),
+} );
```

Comment:
I applaud this initiative, but this schema is missing a couple of top-level items (`sentryUserId` and `lastSeenVersion`, for example).

The best-case scenario is that we have a shared schema for the user-config file between Studio and the CLI. However, to avoid blowing the scope of this PR, I suggest keeping a schema that validates only the `snapshots` array in the user config. All other properties should be left untouched between reading and writing the file.